From ab254364868240623b6643d80a7447d6ee7b0e44 Mon Sep 17 00:00:00 2001
From: Kiran Krishnappa <kiraank@gmail.com>
Date: Wed, 9 Oct 2013 23:30:09 -0700
Subject: [PATCH 100/103] sndfile-util: fix format for 24bit depth wav files

PA_SAMPLE_24NE generated in pa_sndfile_read_sample_spec is not
handled in pa_sndfile_readf and writef function. paplay/parecord
used to get aborted for 24bit depth wav files
---
 src/pulsecore/sndfile-util.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pulsecore/sndfile-util.c b/src/pulsecore/sndfile-util.c
index 0820ee4..a951c01 100644
--- a/src/pulsecore/sndfile-util.c
+++ b/src/pulsecore/sndfile-util.c
@@ -384,6 +384,7 @@ pa_sndfile_readf_t pa_sndfile_readf_function(const pa_sample_spec *ss) {
 
         case PA_SAMPLE_ULAW:
         case PA_SAMPLE_ALAW:
+        case PA_SAMPLE_S24NE:
             return NULL;
 
         default:
@@ -407,6 +408,7 @@ pa_sndfile_writef_t pa_sndfile_writef_function(const pa_sample_spec *ss) {
 
         case PA_SAMPLE_ULAW:
         case PA_SAMPLE_ALAW:
+        case PA_SAMPLE_S24NE:
             return NULL;
 
         default:
-- 
1.8.4

